tree/tree/src/TBranch.cxx

```diff
@@ -1446,6 +1454,66 @@ TFile* TBranch::GetFile(Int_t mode)
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+/// Drops the cluster two behind the current cluster and returns a fresh basket
+/// by either reusing or creating a new one
+
+TBasket* TBranch::GetFreshCluster()
+{
+   TBasket *basket = 0;
+
+   // If GetClusterIterator is called with a negative entry then GetStartEntry will be 0
+   // So we need to check if we reach the zero before we have gone back two clusters
+   // if this is the case, we want to keep everything in memory so we return a new basket
+   TTree::TClusterIterator iter = fTree->GetClusterIterator(fBasketEntry[fReadBasket]);
+   if (iter.GetStartEntry() == 0) return fTree->CreateBasket(this);
+
+   for (Int_t j = 0; j < (fTree->GetMaxVirtualSize() * (-1)); j++) {
+      iter = fTree->GetClusterIterator(iter.GetStartEntry() - 1);
+      if (iter.GetStartEntry() == 0) return fTree->CreateBasket(this);
+   }
+
+   Int_t entryToFlush = fTree->GetClusterIterator(iter.GetStartEntry() - 1).GetStartEntry();
```

Comment:
> to me this seems pretty much the same thing,

Not quite :).  GetClusterIterator has to make a binary search into an array which size is of the order of the number of merged files used to create the current file.  While the iterator has already the correct location in that array (So Prev ought to be a constant operation).